src/drivers/ins/microstrain/MicroStrain.cpp

```diff
@@ -0,0 +1,623 @@
+				break;
+			}
+		}
+
+		if (!is_connected) {
+			_is_init_failed = true;
+			PX4_WARN("Could not connect to the device, exiting");
+			return;
+		}
+	}
+
+
+	PX4_INFO("mip_base_set_idle");
+
+	if (mip_base_set_idle(&device) != MIP_ACK_OK) {
+		PX4_ERR("ERROR: Could not set the device to idle!");
+		return;
+	}
+
+	PX4_INFO("Setting the baud to desired baud rate");
+
+	usleep(500_ms);
+
+	if (mip_3dm_write_uart_baudrate(&device, DESIRED_BAUDRATE) != MIP_ACK_OK) {
+		PX4_ERR("ERROR: Could not set the baudrate!");
+		_is_init_failed = true;
+		return;
+	}
+
+	tcflush(device_uart.uart_get_fd(), TCIOFLUSH);
+
+	usleep(500_ms);
+
+	for (int i = 0; i < 10; i++) {
+		PX4_INFO("Connection Attempt: %d", i);
+
+		if (connect_at_baud(DESIRED_BAUDRATE) == PX4_OK) {
+			break;
+		}
+
+		if (i >= 9) {
+			PX4_ERR("ERROR: Could not reconnect at desired baud!");
+			_is_init_failed = true;
+			return;
+		}
+	}
+
+	if (get_supported_descriptors() != MIP_ACK_OK) {
+		PX4_INFO("ERROR: Could not get descriptors");
+		_is_init_failed = true;
+		return;
+	}
+
+
+	if (configure_imu_message_format() != MIP_ACK_OK) {
+		PX4_INFO("ERROR: Could not write message format");
+		_is_init_failed = true;
+		return;
+	}
+
+
+	//
+	// Register data callbacks
+	//
+	mip_interface_register_packet_callback(&device, &sensor_data_handler, MIP_SENSOR_DATA_DESC_SET, false, &sensor_callback,
+					       this);
+
+
+	//
+	// Setup the rotation based on PX4 standard rotation sets
+	//
+
+	if (mip_3dm_write_sensor_2_vehicle_transform_euler(&device, math::radians<float>(rot_lookup[rotation].roll),
+			math::radians<float>(rot_lookup[rotation].pitch),
+			math::radians<float>(rot_lookup[rotation].yaw)) != MIP_ACK_OK) {
+		PX4_ERR("ERROR: Could not set sensor-to-vehicle transformation!");
+		return;
+	}
+
+
+	if (mip_3dm_write_datastream_control(&device, MIP_3DM_DATASTREAM_CONTROL_COMMAND_ALL_STREAMS, true) != MIP_ACK_OK) {
+		PX4_ERR("ERROR: Could not enable the data stream");
+		return;
+	}
+
+	//
+	//Resume the device
+	//
+
+	if (mip_base_resume(&device) != MIP_ACK_OK) {
+		PX4_ERR("ERROR: Could not resume the device!");
+		return;
+	}
+
+	_is_initialized = true;
+
+}
+
+void MicroStrain::sensor_callback(void *user, const mip_packet *packet, mip::Timestamp timestamp)
+{
+	MicroStrain *ref = static_cast<MicroStrain *>(user);
+
+	if (mip_packet_descriptor_set(packet) != MIP_SENSOR_DATA_DESC_SET) {
+		return;
+	}
+
+	mip_sensor_scaled_accel_data accel;
+	mip_sensor_scaled_gyro_data gyro;
+	mip_sensor_scaled_mag_data mag;
+	mip_sensor_scaled_pressure_data baro;
+
+	bool accel_valid = false;
+	bool gyro_valid = false;
+	bool mag_valid = false;
+	bool baro_valid = false;
+
+	auto t = hrt_absolute_time();
+
+	for (mip_field field = mip_field_first_from_packet(packet); mip_field_is_valid(&field); mip_field_next(&field)) {
+		switch (mip_field_field_descriptor(&field)) {
+		case MIP_DATA_DESC_SENSOR_ACCEL_SCALED: extract_mip_sensor_scaled_accel_data_from_field(&field, &accel);
```

Comment:
All these cases should be indented, and everything after the `:` should be on the next line indented as well. so this case would become
```
case MIP_DATA_DESC_SENSOR_ACCEL_SCALED:
    extract_mip_sensor_scaled_accel_data_from_field(&field, &accel);
    accel_valid = true;
    break;
```